File: concept/17_recursion/linear_search/linear_search.dart
import 'dart:io';

bool sum(List<int> arr, int size, int key) {
  if (size == 0) {
    return false;
  }

  if (arr[0] == key) {
    return true;
  }

  bool big = sum(arr.skip(1).toList(), size - 1, key);

  return big;
}

void main() {
  print("Enter key to search");
  int key = int.parse(stdin.readLineSync()!);
  List<int> arr = [3, 2, 5, 1, 6];

  bool result = sum(arr, arr.length, key);
  if (result) {
    print("Present");
  } else {
    print("absent");
  }
}

File: concept/17_recursion/linear_search/linear_search.cpp
#include <iostream>
using namespace std;

bool search(int arr[], int size, int key)
{
    if (size == 0)
    {
        return false;
    }

    if (arr[0] == key)
    {
        return true;
    }
    else
    {
        bool big = search(arr + 1, size - 1, key);
        return big;
    }
}

int main()
{

    cout << "Enter key to search" << endl;
    int key;
    cin >> key;
    int arr[5] = {
        3,
        2,
        5,
        1,
        6};

    bool result = search(arr, 5, key);
    if (result)
    {

        cout << "true" << endl;
    }
    else
    {
        cout << "false" << endl;
    }
}